apps/account-api/src/services/siwfV2.service.ts

```diff
@@ -177,12 +181,9 @@ export class SiwfV2Service {
 
     response.controlKey = payload.userPublicKey.encodedValue;
 
-    // Try to look up the MSA id, if there is no createSponsoredAccountWithDelegation request
-    if (payload.payloads.every((x) => x.endpoint?.extrinsic !== 'createSponsoredAccountWithDelegation')) {
-      // Get the MSA Id from the chain
-      const msaId = await this.blockchainService.publicKeyToMsaId(response.controlKey);
-      if (msaId) response.msaId = msaId;
-    }
+    // Get the MSA Id from the chain, if it exists
+    const msaId = await this.blockchainService.publicKeyToMsaId(response.controlKey);
+    if (msaId) response.msaId = msaId;
```

Comment:
Because it's possible the same payload could be submitted multiple times, we should check for the MSA anyway. (It doesn't hurt; only costs an RPC query to the chain)